wandb/sdk/artifacts/artifact_saver.py

```diff
@@ -109,7 +113,7 @@ def _save_internal(
         incremental: bool = False,
         history_step: Optional[int] = None,
         base_id: Optional[str] = None,
-    ) -> Optional[Dict]:
+    ) -> Tuple[Optional[Dict], Optional[concurrent.futures.Future]]:
```

Comment:
Nit: not sure why the return type here was `Optional`, we never return `None`.

Reply:
good catch, The dict cannot be None but the future could, so I tighten up the types here a bit